modules/contact/src/ContactAction.C

```diff
@@ -90,7 +90,8 @@ ContactAction::act()
       InputParameters params = _factory.getValidParams("MechanicalContactConstraint");
 
       // Extract global params
-      _app.parser().extractParams(_name, params);
+      if (isParamValid("parser_syntax"))
+        _app.parser().extractParams(getParam<std::string>("parser_syntax"), params);
```

Comment:
This isn't quite equivalent. It's very unlikely that parser_syntax wouldn't exist but with your change, if it didn't this action would cease to work. I think it would be better to error out if it doesn't exist.